gossip/api/types.go

```diff
@@ -0,0 +1,229 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package api describes the external interface to the gossip hub.
+package api
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/google/certificate-transparency-go/tls"
+)
+
+// PathPrefix is the common prefix for all HTTP entrypoints.
+const PathPrefix = "/gossip/v0/"
+
+// Add Log Head to Hub
+// POST https://<hub server>/gossip/v0/add-log-head
+// Inputs:
+//    source_url: The URL from which the log head was obtained
+//    head_data: The binary-encoded head
+//    log_signature: The source log's signature over head_data.
+// Outputs:
+//    timestamped_entry: TLS-encoded TimestampedEntry.
+//    hub_signature: hub's signature gossip_timestamped_entry
+
+// AddLogHeadPath is the final path component for this entrypoint.
+const AddLogHeadPath = "add-log-head"
+
+// AddLogHeadRequest represents the JSON request body sent to the add-log-head POST method.
+type AddLogHeadRequest struct {
+	SourceURL    string `json:"source_url"`
```

Comment:
I think we were also thinking of including a `LogID` and optional `ConsistencyProof per
https://docs.google.com/document/d/1i4cS9FNiddBkm5ZD3F5DYHkykiaf-jeATFMjtvAD7Jc/edit#

Reply:
At the moment, I'm thinking of this as a generic repository of signed blobs -- I've actually gone back and dropped mention of Logs as much as possible from the API (and `LogID` is very CT-Log-specific).  

ConsistencyProof would be specific to signed-blobs-from-a-Merkle tree, but if it's optional that's not a problem. 

(BTW, that's not a public doc, so probably best not to reference it here)

Reply:
`LogID` is equivalent to a `KeyID` which helps with signature verification when there are many public keys to try.   Something like it would still be needed if the monitor is just a log of signed blobs.